Add mutability.md tag (#858)
python-discord/bot: bot/resources/tags/mutability.md

**Mutable vs immutable objects**

Imagine that you want to make all letters in a string upper case. Conveniently, strings have an `.upper()` method.

You might think that this would work:
```python
string = "abcd"
string.upper()

> print(string) # abcd

Reviewer: Would it perhaps be better to show an interactive session here?
```python
>>> string = "abcd"
>>> string.upper()
'ABCD'
>>> string
'abcd'
```
This demonstrates that the new string is returned, and needs to be assigned to `string`.

Reviewer 2: Also, should `string` be renamed to `text` instead, to avoid shadowing the `string` module?

Reviewer: I agree, this would be less confusing.

Author: changed

```

`string` didn't change. Why is that so?

> That's because strings in Python are _immutable_. You can't change them, you can only pass around existing strings or create new oness.

Reviewer: ```suggestion
That's because strings in Python are _immutable_. You can't change them, you can only pass around existing strings or create new ones.
```

```python
string = "abcd"
string = string.upper()
```

> `string.upper()` creates a new string which is like the old one, but with allthe letters turned to upper case.

Reviewer: ```suggestion
`string.upper()` creates and returns a new string which is like the old one, but with all the letters turned to upper case.
```

`int`, `float`, `complex`, `tuple`, `frozenset` are other examples of immutable data types in Python.

Mutable data types like `list`, on the other hand, can be changed in-place:
```python
my_list = [1, 2, 3]
my_list.append(4)
print(my_list) # [1, 2, 3, 4]
```

> `dict` and `set` are other examples of mutable data types in Python.

Reviewer: ```suggestion
Other examples of mutable data types in Python are `dict` and `set`.
```
Just a minor nit-pick, but I don't like that this sentence is a copy of the one on line 23. Reversing it like this would remedy that, but it's not necessary.

**Mutable vs immutable objects**

Imagine that you want to make all letters in a string upper case.
Conveniently, strings have an `.upper()` method.

You might think that this would work:
```python
string = "abcd"
string.upper()
print(string) # abcd
```

`string` didn't change. Why is that so?

That's because strings in Python are _immutable_. You can't change them, you can only pass
around existing strings or create new ones.

```python
string = "abcd"
string = string.upper()
```
`string.upper()` creates a new string which is like the old one, but with all
the letters turned to upper case.

> `int`, `float`, `complex`, `tuple`, `frozenset`  are other examples of immutable data types in Python.

Reviewer 3: ```suggestion
`int`, `float`, `complex`, `tuple`, `frozenset` are other examples of immutable data types in Python.
```

Author: Fixed hard-wrapping.
